ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/inmemory/InMemoryLedger.scala

```diff
@@ -272,21 +259,39 @@ class InMemoryLedger(
       acs.parties.values.toList
     })
 
-  override def allocateParty(
+  override def publishPartyAllocation(
+      submissionId: SubmissionId,
       party: Party,
-      displayName: Option[String]): Future[PartyAllocationResult] =
-    Future.successful(this.synchronized {
+      displayName: Option[String]): Future[SubmissionResult] =
+    Future.successful(this.synchronized[SubmissionResult] {
```

Comment:
Do you think it would be possible to reduce the scope of the lock, maybe to something private?

Reply:
Maybe. I wouldn't make this change in this PR either way though.